src/components/Main/Main.jsx

```diff
@@ -0,0 +1,252 @@
+import React, { useEffect, useState } from "react";
+import Papaparse from "papaparse"
+import searchIcon from "../../assets/search-icon.svg"
+import downloadIcon from "../../assets/download-icon.svg"
+import './styles.scss'
+import About from "../About/About";
+import SortArrows from "../SortArrows/SortArrows";
+
+const titles = ["date", "company", "position", "salary", "years of experience"];
+let copyOfRows = [];
+
+// const [rows, setRows] = useState([])
+// useEffect(() => {
+//   fetch("./data.csv")
+//     .then(res => res.text())
+//     .then(csv => {
+//       const json = Papaparse.parse(csv)
+//       json.data.shift()
+//       console.log("json.data", json.data)
+//       setRows(json.data)
+//     })
+// }, [])
+
+const Main = () => {
+
+    const [rows, setRows] = useState({});
+    const [sortType, setSortType] = useState();
+
+    useEffect(() => {
+        fetch('./data.csv')
+            .then(res => res.text())
+            .then(csv => {
+                const json = Papaparse.parse(csv)
+                setRows(transformData(json.data))
+                copyOfRows = transformData(json.data)
+            })
+    }, [])
+
+    const transformData = (payload) => {
+        const headers = payload[0];
+        const [first, ...data] = payload;
+
+        const transformedData = data.map((item, index) => {
+            let obj = {};
+            item.forEach((i, idx) => {
+                obj[headers[idx]] = i;
+            })
+            return obj;
+        });
+
+        return transformedData;
+    }
+
+    const sortTable = (arg) => {
+        if (arg == "dateUp") {
+            rows.sort((a, b) => new Date(b["Date"]) - new Date(a["Date"]));
+        }
+        else if (arg == "dateDown") {
+            rows.sort((a, b) => new Date(a["Date"]) - new Date(b["Date"]));
+        }
+        else if (arg == "companyUp") {
```

Comment:
First of all, when you call JS sort function, it will change main value(In this case rows, but in react you can change local state with setState function(using second argument of useState hook), in this case you can use another hook which you prefer "useCallback" or "useMemo")

It will be perfect to have global sorting function which will receive "sort by" and will return sorted array as result.